indexing-service/src/main/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTask.java

```diff
@@ -225,23 +226,33 @@ public TaskStatus runTask(TaskToolbox toolbox) throws Exception
       final Set<Interval> unusedSegmentIntervals = unusedSegments.stream()
                                                                  .map(DataSegment::getInterval)
                                                                  .collect(Collectors.toSet());
-      // Fetch the load specs of all segments overlapping with the given interval
-      final Set<Map<String, Object>> usedSegmentLoadSpecs = toolbox
-          .getTaskActionClient()
-          .submit(new RetrieveUsedSegmentsAction(getDataSource(), null, unusedSegmentIntervals, Segments.INCLUDING_OVERSHADOWED))
-          .stream()
-          .map(DataSegment::getLoadSpec)
-          .collect(Collectors.toSet());
+      final Set<Map<String, Object>> usedSegmentLoadSpecs = new HashSet<>();
+      if (!unusedSegmentIntervals.isEmpty()) {
+        // Fetch the load specs of all segments overlapping with the given interval
+        usedSegmentLoadSpecs.addAll(toolbox.getTaskActionClient()
+                                           .submit(new RetrieveUsedSegmentsAction(
+                                               getDataSource(),
+                                               null,
+                                               unusedSegmentIntervals,
+                                               Segments.INCLUDING_OVERSHADOWED
+                                           ))
+                                           .stream()
+                                           .map(DataSegment::getLoadSpec)
+                                           .collect(Collectors.toSet())
+        );
+      }
 
       // Kill segments from the deep storage only if their load specs are not being used by any used segments
       final List<DataSegment> segmentsToBeKilled = unusedSegments
           .stream()
-          .filter(unusedSegment -> !usedSegmentLoadSpecs.contains(unusedSegment.getLoadSpec()))
+          .filter(unusedSegment -> !usedSegmentLoadSpecs.contains(unusedSegment.getLoadSpec())
+                                   || unusedSegment.getLoadSpec() == null)
```

Comment:
unusedSegment.getLoadSpec() == null should be checked first IMO.

Reply:
Done